File: 2501-longest-square-streak-in-an-array/2501-longest-square-streak-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSquareStreak = function(nums) {
  let streak = -1;
  const set = new Set(nums);

  for (let i = 0; i < nums.length; i++) {
    let curr = nums[i];
    let length = 0;
    while (set.has(curr)) {   
      length++;
      curr = curr ** 2;
    }
    
    if (length > 1) {
      streak = Math.max(streak, length);      
    }
  }

  return streak;
};

File: 2501-longest-square-streak-in-an-array/2501-longest-square-streak-in-an-array.py
class Solution(object):
    def longestSquareStreak(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        streak = -1;
        unique = set(nums);

        i = 0
        while i < len(nums):
            curr = nums[i]
            length = 0
            while curr in unique:
                length += 1
                curr = curr ** 2

            if length > 1:
                streak = max(streak, length)        
            i += 1

        return streak
        